exchange/auction.go

```diff
@@ -78,7 +81,7 @@ func (a *auction) doCache(ctx context.Context, cache prebid_cache_client.Client,
 					toCache = append(toCache, prebid_cache_client.Cacheable{
 						Type:       prebid_cache_client.TypeJSON,
 						Data:       jsonBytes,
-						TTLSeconds: cacheTTL(expByImp[impID], topBidPerBidder.bid.Exp, ttlBuffer),
+						TTLSeconds: cacheTTL(expByImp[impID], topBidPerBidder.bid.Exp, defTTLByImp[impID], ttlBuffer),
```

Comment:
As written, I only use the default if neither bidTTL or impTTL are defined (defined means `>0`, just in case anyone gets creative and tries setting a negative ttl). I use as default the max ttl among the formats found in `Imp`

This is why I have a separate `expByImp` and `defTTLByImp`, I want `min(bidTTL, impTTL)`, but I don't want the default to shorten the TTL below either of these. So I need to see the case where `imp.exp` is not set, and the default is shorter than `bidTTL`. Otherwise I could just use the default as the vale for `expByImp` and save a step. :)

I missed that we had `bidType` for the bids, that does simplify things a bit.